src/tests/system/tests/test_authentication.py

```diff
@@ -97,3 +98,45 @@ def test_authentication__offline_login(client: Client, provider: GenericProvider
 
     assert client.auth.parametrize(method).password(user, correct), "offline login with correct password failed"
     assert not client.auth.parametrize(method).password(user, wrong), "offline login with wrong password succeeded"
+
+
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(gh=7174)
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_authentication__login_using_email_address(client: Client, ad: AD, method: str, sssd_service_user: str):
+    """
+    :title: Login using user's email address
+    :description:
+        Testing the feature to login using an email address instead of the userid. The email can be anything and does
+        not have to be derived from the domain name an alias can be used. The email should be case insensitive and
+        permit special characters. For example "user_3@, usER_3@" should both work.
+    :setup:
+        1. Add AD users with different email addresses
+        2. Start SSSD
+    :steps:
+        1. Authenticate users using their email address and in different cases
+    :expectedresults:
+        1. Authentication is successful using the email address and is case insensitive
+    :customerscenario: False
+    """
+    ad.user("user-1").add(password="Secret123", email=f"user-1@{ad.host.domain}")
+    ad.user("user-2").add(password="Secret123", email="user-2@alias-domain.com")
+    ad.user("user-3").add(password="Secret123", email="user_3@alias-domain.com")
+
+    client.sssd.set_service_user(sssd_service_user)
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(
+        f"user-1@{ad.host.domain}", "Secret123"
+    ), "login with correct password failed"
+    assert client.auth.parametrize(method).password(
+        "user-2@alias-domain.com", "Secret123"
+    ), "login with correct password failed"
+    assert client.auth.parametrize(method).password(
+        "uSEr_3@alias-domain.com", "Secret123"
```

Comment:
The test looks great, thank you.

I think that it should also work with domain part case-insensitive. Can you make that mail "uSEr_3@alias-dOMain.com"?

Reply:
Done.